.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: RSpec CI
+on:
+  push:
+    branches:
+      - 'main'
+      - '*-maintenance'
+      - '*-dev'
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  test:
+    name: 'Ruby: ${{ matrix.ruby }}, Rails: ${{ matrix.env.RAILS_VERSION }}'
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+       include:
+         # Rails 6.0 builds >= 2.5.0
+         - ruby: 3.0.0-preview1
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.7
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.6
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.5
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         # Rails 5.2 builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.2.10
+           allow_failure: true
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '5-2-stable'
+         # Rails 5.1 Builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         # Rails 5.0 Builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+    env: ${{ matrix.env }}
+    continue-on-error: ${{ matrix.allow_failure || false }}
```

Comment:
I doesn't seem to work. :/
https://github.com/rspec/rspec-rails/runs/1438441078?check_suite_focus=true

Reply:
```
SyntaxError: /home/runner/work/rspec-rails/bundle/ruby/2.2.0/gems/actionpack-5.2.4.4/lib/action_dispatch/request/session.rb:96: syntax error, unexpected '.'
          id&.public_id
```
That build is failing genuinely.

Reply:
I think they introduced a 2.2.x bug for 5.2... `&.` is 2.3 and above.

Reply:
We have this in 3-9-maintenance https://github.com/rspec/rspec-rails/commit/6a8fd077f14c791a7ef39f596735a7d867335a9e#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485R58

Reply:
Oh, sorry, I've missed 2.2.2 is tested against 5-2-stable already.

Reply:
> I think they introduced a 2.2.x bug for 5.2... &. is 2.3 and above.

Yes completely and I didn't see a patch to change that ATM.